components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/js/JsClaims.java

```diff
@@ -87,6 +89,46 @@ public void initializeContext(AuthenticationContext context) {
         super.initializeContext(context);
         if (StringUtils.isNotBlank(idp) && getContext().getCurrentAuthenticatedIdPs().containsKey(idp)) {
             this.authenticatedUser = getContext().getCurrentAuthenticatedIdPs().get(idp).getUser();
+        } else {
+            this.authenticatedUser = getAuthenticatedUserFromSubjectIdentifierStep();
+        }
+    }
+
+    /**
+     * Get authenticated user from step config of current subject identifier.
+     *
+     * @return AuthenticatedUser.
+     */
+    private AuthenticatedUser getAuthenticatedUserFromSubjectIdentifierStep() {
+
+        AuthenticatedUser authenticatedUser = null;
+        StepConfig stepConfig = getCurrentSubjectIdentifierStep();
+        if (stepConfig != null) {
+            authenticatedUser = getCurrentSubjectIdentifierStep().getAuthenticatedUser();
+        }
+        return authenticatedUser;
+    }
+
+    /**
+     * Retrieve step config of current subject identifier.
+     *
+     * @return StepConfig.
+     */
+    private StepConfig getCurrentSubjectIdentifierStep() {
+
+        if (getContext().getSequenceConfig() == null) {
+            // Sequence config is not yet initialized.
+            return null;
+        }
+        Map<Integer, StepConfig> stepConfigs = getContext().getSequenceConfig().getAuthenticationGraph().getStepMap();
+        Optional<StepConfig> subjectIdentifierStep = stepConfigs.values().stream()
+                .filter(stepConfig -> (stepConfig.isCompleted() && stepConfig.isSubjectIdentifierStep())).findFirst();
+        if (subjectIdentifierStep.isPresent()) {
+            return subjectIdentifierStep.get();
+        } else if (getContext().getCurrentStep() > 0) {
+            return stepConfigs.get(getContext().getCurrentStep());
+        } else {
+            return null;
```

Comment:
This will set the authenticated user as a null. Is that fine? For example, then the [1] and [2] will return null and false.
[1] public Object getMember(String claimUri)
[2] public boolean hasMember(String claimUri)

Reply:
this method will be called, if there is an already existing session for an AuthenticatedUser. which means that user will already have a subject identifier step executed in previous session. There AFAIU if the subject identifier step is null , it would be okay to return null object.. WDYT?